src/test/scala-2.11/com/linagora/gatling/imap/scenario/ImapFullScenario.scala

```diff
@@ -0,0 +1,67 @@
+package com.linagora.gatling.imap.scenario
+
+import java.util.Calendar
+
+import com.linagora.gatling.imap.PreDef._
+import com.linagora.gatling.imap.protocol.Uid
+import com.linagora.gatling.imap.protocol.command.FetchAttributes.AttributeList
+import com.linagora.gatling.imap.protocol.command.FetchRange._
+import io.gatling.app.Gatling
+import io.gatling.core.Predef._
+import io.gatling.core.config.GatlingPropertiesBuilder
+import io.gatling.core.scenario.Simulation
+
+import scala.collection.immutable.Seq
+import scala.concurrent.duration._
+
+class ImapFullScenario extends Simulation {
+  val feeder = csv("users.csv").circular
+  val userCount: Int = feeder.records.length
+
+  val receiveEmail = exec(imap("append").append("INBOX", Some(Seq("\\Flagged")), Option.empty[Calendar],
+                  """From: expeditor@example.com
+                    |To: recipient@example.com
+                    |Subject: test subject
+                    |
+                    |Test content
+                    |abcdefghijklmnopqrstuvwxyz
+                    |0123456789""".stripMargin).check(ok))
+
+  val readLastEmail = exec(imap("list").list("", "*").check(ok, hasFolder("INBOX")))
+                      .exec(imap("select").select("INBOX").check(ok))
+                      .exec(imap("fetch").fetch(Seq(Last()), AttributeList("BODY[HEADER]", "UID", "BODY[TEXT]")).check(ok))
+
+  val heavyUser = repeat(3)(receiveEmail)
+                  .repeat(2)(readLastEmail)
+  val lightUser = receiveEmail
+                  .exec(readLastEmail)
+
+  val duration = 4 hours
+
+  val scn = scenario("Imap").feed(feeder)
+    .pause(1 second)
+    .exec(imap("Connect").connect()).exitHereIfFailed
+    .exec(imap("login").login("${username}", "${password}").check(ok))
+    .exec {
+      randomSwitch(
+        75.0 -> during(duration) {
+                  pace(5 minutes, 15 minutes)
+                  .exec(lightUser)
+                },
+
+        25.0 -> during(duration) {
+                  pace(30 seconds, 90 seconds)
+                  .exec(heavyUser)
+                }
+      )
+
+    }
+
+  val userCountWith50PercentsOfThemAlsoReadingTheirEmailOnAnotherDevice = (userCount * 1.5).toInt
+
+  setUp(scn.
+      inject(
+          nothingFor(1 seconds),
+          rampUsers(userCountWith50PercentsOfThemAlsoReadingTheirEmailOnAnotherDevice) over(10 minutes)))
```

Comment:
ok for the ramp.
But for the duration, it's not exactly what you describe because you don't constraint the time, you just do some stuff for some time then do some handmade calculation to evaluate the duration. Here, the exact time is 10 minutes + time for connect + login + 4 hours + time for 3 receiveEmail + 2 readlastemail. It's not a big deal, but just adding a maxDuration will make this deterministic.
That's the same for the load : here the load depends a lot on pace+ramp. You could make sure to manage the max load with a throttle clause.